src/main/plugins/org.dita.base/xsl/preprocess/maprefImpl.xsl

```diff
@@ -156,8 +156,14 @@ See the accompanying LICENSE file for applicable license.
                 </xsl:otherwise>
               </xsl:choose>
             </xsl:variable>
-            <xsl:variable name="targetTitleAndTopicmeta" as="element()*"
-              select="$file/*/*[contains(@class,' topic/title ') or contains(@class,' map/topicmeta ')]"/>
+            <xsl:variable name="targetTitleAndTopicmeta" as="element()*">
+              <!-- submap title -->
+              <xsl:sequence select="$file/*/*[contains(@class,' topic/title ')]"/>
+              <!-- metadata in mapref -->
+              <xsl:sequence select="*[contains(@class,' map/topicmeta ')]"/>
```

Comment:
This looks pretty simple / straightforward. Couple of questions:
1. Do we have any way to distinguish between the mapref `topicmeta` and the submap `topicmeta`? There could be a case where I want to do something with submap metadata (like I did at one point when doing the original bit here, but *not* want to do that with mapref metadata. I think with this, if we had only mapref metadata, I would think it came from the map. Kind of hard to come up with a reason where that might matter but it's possible - maybe alt-titles on a mapref would throw things off if present in a submap?
2. Looking at where this is processed, we pass in `relative-path` based on the path of the referenced map. I think this means that any relative links in the topicmeta for the mapref, will be updates as if they are located in the submap. If the mapref and submap are in the same directory, no issue, but if the submap is in a subdirectory, metadata like `<source href="../somefile.dita"/>` would fail. Definitely uncommon but something to consider

Reply:
@robander - you raise good points.

For #1, indeed there is currently no way to differentiate between submap (`3:`) and mapref (`4:`) metadata:

```
<submap>
   <submap-topicmeta>
      <submap-title>My Submap</submap-title>
   </submap-topicmeta>
   <submap-topicmeta-container>
      <shortdesc>3:topicmeta/shortdesc</shortdesc>
   </submap-topicmeta-container>
   <submap-topicmeta-container>
      <shortdesc>4:topicmeta/shortdesc</shortdesc>
   </submap-topicmeta-container>
   <topicref href="topic.dita">
      <topicref href="subtopic.dita"/>
   </topicref>
</submap>
```

How about if I put mapref metadata in its own `<submap-mapref-topicmeta-container>` element?

```
<submap>
   <submap-topicmeta>
      <submap-title>My Submap</submap-title>
   </submap-topicmeta>
   <submap-topicmeta-container>
      <shortdesc>3:topicmeta/shortdesc</shortdesc>
   </submap-topicmeta-container>
   <submap-mapref-topicmeta-container>  <!-- note the "-mapref" -->
      <shortdesc>4:topicmeta/shortdesc</shortdesc>
   </submap-mapref-topicmeta-container>
   <topicref href="topic.dita">
      <topicref href="subtopic.dita"/>
   </topicref>
</submap>
```

This allows for filtering, precedence, and aggregation via XPath. I think I like this approach. If you are also agreeable, I'll update the pull request.

For #2, I will need to look at the code some more and try to sensitize it with a testcase. Would you be agreeable to handling that through a follow-up issue and pull request?

Reply:
For item 1 -- that sounds like a good solution, yes. (Note that the container today is a pseudo-specialization of `foreign`, which should also be the case for `submap-mapref-topicmeta-contaner`). Typing that out, it could also be shortened to `mapref-topicmeta-container`...

For item 2 - yes, it could be a follow-up. This is a pretty extreme edge case, and the same issue with paths probably still exists in some dark corners of the code.

Reply:
@robander - I force-pushed the updated code. I think this is much better, thank you! In this implementation, issue 2 shouldn't be an issue because the current map's file context remains in place when resolving the mapref topicmeta.

I know you had concerns about end-of-line comments [here](https://github.com/dita-ot/dita-ot/pull/4176#discussion_r1177968948), but I see many such existing comments so I decided to sneak some more in...